File: backend/internal/model/dto/reclamo.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

// CreateReclamoRequest formulario público de INDECOPI.
type CreateReclamoRequest struct {
	TipoSolicitud string `json:"tipo_solicitud" validate:"required,oneof=RECLAMO QUEJA"`

	// Sección 1: Consumidor
	NombreCompleto  string `json:"nombre_completo" validate:"required"`
	TipoDocumento   string `json:"tipo_documento" validate:"required,oneof=DNI CE Pasaporte RUC"`
	NumeroDocumento string `json:"numero_documento" validate:"required"`
	Telefono        string `json:"telefono" validate:"required"`
	Email           string `json:"email" validate:"required,email"`
	Domicilio       string `json:"domicilio"`
	Departamento    string `json:"departamento"`
	Provincia       string `json:"provincia"`
	Distrito        string `json:"distrito"`
	MenorDeEdad     bool   `json:"menor_de_edad"`
	NombreApoderado string `json:"nombre_apoderado"`

	// Sección 3: Bien contratado
	TipoBien        string  `json:"tipo_bien" validate:"omitempty,oneof=PRODUCTO SERVICIO"`
	MontoReclamado  float64 `json:"monto_reclamado"`
	DescripcionBien string  `json:"descripcion_bien" validate:"required"`
	NumeroPedido    string  `json:"numero_pedido"`

	// Campos QUEJA
	AreaQueja            string `json:"area_queja"`
	DescripcionSituacion string `json:"descripcion_situacion"`

	// Sección 4: Detalle
	FechaIncidente   string `json:"fecha_incidente" validate:"required"`
	DetalleReclamo   string `json:"detalle_reclamo" validate:"required"`
	PedidoConsumidor string `json:"pedido_consumidor" validate:"required"`

	// Firma
	FirmaDigital string `json:"firma_digital"`

	// Sede (desde la URL pública)
	SedeSlug string `json:"sede_slug"`

	// Validación cliente empresa
	EsClienteRegistrado bool `json:"es_cliente_registrado"`

	// Archivos adjuntos (claves de Cloudflare R2)
	ArchivosAdjuntos []string `json:"archivos_adjuntos"`

	// CAPTCHA Cloudflare Turnstile
	TurnstileToken string `json:"turnstile_token"`
}

// ReclamoListItem resumen para la tabla del dashboard.
type ReclamoListItem struct {
	ID                   uuid.UUID  `json:"id"`
	CodigoReclamo        string     `json:"codigo_reclamo"`
	TipoSolicitud        string     `json:"tipo_solicitud"`
	Estado               string     `json:"estado"`
	NombreCompleto       string     `json:"nombre_completo"`
	Email                string     `json:"email"`
	FechaRegistro        time.Time  `json:"fecha_registro"`
	FechaLimiteRespuesta *time.Time `json:"fecha_limite_respuesta"`
	SedeNombre           *string    `json:"sede_nombre"`
	DiasRestantes        *int       `json:"dias_restantes"`
	Prioridad            string     `json:"prioridad"`
}

// UpdateEstadoRequest cambio de estado.
type UpdateEstadoRequest struct {
	Estado     string `json:"estado" validate:"required,oneof=PENDIENTE EN_PROCESO RESUELTO CERRADO RECHAZADO"`
	Comentario string `json:"comentario"`
}

// AsignarReclamoRequest asignación a un admin.
type AsignarReclamoRequest struct {
	AdminID uuid.UUID `json:"admin_id" validate:"required"`
}

// ReclamoTrackingResponse respuesta para el seguimiento público.
type ReclamoTrackingResponse struct {
	CodigoReclamo        string     `json:"codigo_reclamo"`
	Estado               string     `json:"estado"`
	FechaRegistro        time.Time  `json:"fecha_registro"`
	FechaLimiteRespuesta *time.Time `json:"fecha_limite_respuesta"`
	FechaRespuesta       *time.Time `json:"fecha_respuesta"`
	SedeNombre           *string    `json:"sede_nombre"`
	TipoSolicitud        string     `json:"tipo_solicitud"`
	DescripcionBien      string     `json:"descripcion_bien"`
	RespuestaEmpresa     string     `json:"respuesta_empresa,omitempty"`
}

// PublicMessageRequest mensaje enviado desde el seguimiento.
type PublicMessageRequest struct {
	Mensaje       string `json:"mensaje" binding:"required,max=1500"`
	ArchivoURL    string `json:"archivo_url"`
	ArchivoNombre string `json:"archivo_nombre"`
}

File: frontend/src/tipos/reclamo.ts
import type { Nullable } from './api';

export type TipoSolicitud = 'RECLAMO' | 'QUEJA';
export type EstadoReclamo = 'PENDIENTE' | 'EN_PROCESO' | 'RESUELTO' | 'CERRADO' | 'RECHAZADO';
export type CanalOrigen = 'WEB' | 'WHATSAPP' | 'EMAIL' | 'CHATBOT';

export interface Reclamo {
  id: string;
  tenant_id: string;
  codigo_reclamo: string;
  tipo_solicitud: TipoSolicitud;
  estado: EstadoReclamo;

  // Consumidor
  nombre_completo: string;
  tipo_documento: string;
  numero_documento: string;
  telefono: string;
  email: string;
  domicilio: Nullable<string>;
  departamento: Nullable<string>;
  provincia: Nullable<string>;
  distrito: Nullable<string>;
  menor_de_edad: boolean;
  nombre_apoderado: Nullable<string>;

  // Proveedor snapshot
  razon_social_proveedor: Nullable<string>;
  ruc_proveedor: Nullable<string>;
  direccion_proveedor: Nullable<string>;

  // Sede snapshot
  sede_id: Nullable<string>;
  sede_nombre: Nullable<string>;
  sede_direccion: Nullable<string>;

  // Bien contratado
  tipo_bien: Nullable<string>;
  monto_reclamado: Nullable<number>;
  descripcion_bien: string;
  numero_pedido: Nullable<string>;

  // Detalle
  area_queja: Nullable<string>;
  descripcion_situacion: Nullable<string>;
  fecha_incidente: string;
  detalle_reclamo: string;
  pedido_consumidor: string;

  // Firma y metadata
  firma_digital: Nullable<string>;
  ip_address: Nullable<string>;
  user_agent: Nullable<string>;
  acepta_terminos: boolean;
  acepta_copia: boolean;

  // Fechas
  fecha_registro: string;
  fecha_limite_respuesta: Nullable<string>;
  fecha_respuesta: Nullable<string>;
  fecha_cierre: Nullable<string>;

  atendido_por: Nullable<string>;
  canal_origen: CanalOrigen;
  /* Coincide con la columna `reclamos.es_cliente_registrado` del backend.
     Indica si el consumidor ya figuraba como cliente de la empresa segun
     la validacion contra el sistema externo. */
  es_cliente_registrado: boolean;
  nombre_atendido_por?: string;
  
  // Agregado para soportar respuestas de la API
  mensaje?: string;

  // Archivos adjuntos (claves de Cloudflare R2)
  archivos_adjuntos?: string[] | null;

  // --- NUEVOS CAMPOS (Opcionales para no romper) ---
  respuesta_empresa?: string;
  dias_restantes?: number;
}

export interface CrearReclamoRequest {
  tipo_solicitud: TipoSolicitud;
  sede_slug?: string;

  nombre_completo: string;
  tipo_documento: string;
  numero_documento: string;
  telefono: string;
  email: string;
  domicilio?: string;
  departamento?: string;
  provincia?: string;
  distridto?: string;
  menor_de_edad: boolean;
  nombre_apoderado?: string;

  tipo_bien?: string;
  monto_reclamado?: number;
  descripcion_bien: string;
  numero_pedido?: string;

  // Empresa (opcional, solo si es cliente empresa)
  ruc_empresa?: string;
  razon_social_empresa?: string;

  area_queja?: string;
  descripcion_situacion?: string;
  fecha_incidente: string;
  detalle_reclamo: string;
  pedido_consumidor: string;

  firma_digital?: string;
  acepta_terminos: boolean;
  acepta_copia: boolean;

  archivos_adjuntos?: string[];

  // Validación cliente empresa
  es_cliente_registrado?: boolean;

  // CAPTCHA Cloudflare Turnstile
  turnstile_token?: string;
}

export interface CambiarEstadoRequest {
  estado: EstadoReclamo;
  comentario?: string;
}

export interface AsignarReclamoRequest {
  admin_id: string;
}

export interface ReclamoTracking {
  codigo_reclamo: string;
  estado: EstadoReclamo;
  fecha_registro: string;
  fecha_limite_respuesta: string;
  fecha_respuesta?: string;
  sede_nombre?: string;
  tipo_solicitud: TipoSolicitud;
  descripcion_bien: string;
  respuesta_empresa?: string;
}

export const ESTADOS_RECLAMO: Record<EstadoReclamo, { etiqueta: string; color: string }> = {
  PENDIENTE: { etiqueta: 'Pendiente', color: '#a67718' },
  EN_PROCESO: { etiqueta: 'En Proceso', color: '#b85528' },
  RESUELTO: { etiqueta: 'Resuelto', color: '#5c8a4f' },
  CERRADO: { etiqueta: 'Cerrado', color: '#857e70' },
  RECHAZADO: { etiqueta: 'Rechazado', color: '#b83a32' },
};